src/wallet/wallet.cpp

```diff
@@ -970,6 +1004,22 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransactionRef& ptx, const CBlockI
         if (fExisted && !fUpdate) return false;
         if (fExisted || IsMine(tx) || IsFromMe(tx))
         {
+            // check if we need to flag used keypool keys
+            std::set<CKeyID> setKeyPool;
+            GetAllReserveKeys(setKeyPool);
```

Comment:
I guess this is very inefficient and something like #10235 should allow to speed up things.